ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/state/ServerStateDynamicBinding.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.rest.server.state;
+
+import javax.ws.rs.container.DynamicFeature;
+import javax.ws.rs.container.ResourceInfo;
+import javax.ws.rs.core.FeatureContext;
+import javax.ws.rs.ext.Provider;
+
+@Provider
+public class ServerStateDynamicBinding implements DynamicFeature {
+  private final ServerState state;
+
+  public ServerStateDynamicBinding(final ServerState state) {
+    this.state = state;
+  }
+
+  @Override
+  public void configure(final ResourceInfo resourceInfo, final FeatureContext context) {
+    final String packageName = resourceInfo.getResourceClass().getPackage().getName();
+    if (packageName.startsWith("io.confluent.ksql.rest")) {
+      context.register(new ServerStateFilter(state));
```

Comment:
I think you can register this filter directly from the `KsqlRestApplication#configureBaseApplication` method which applies to all REST calls. I've done it in the past, or is there a benefit when using the `DynamicFeature` interface?

Reply:
yeah - I only want the filter to apply for the resources we define, which is what the package name check is doing.

Reply:
That's basically all of the REST methods that exist in KSQL, isn't it? I find this `ServerStateFilter` pretty useful to provide state info to any REST call, even those added dynamically by any resource extension that KSQL calls.

However, this can be out of the scope for this PR. If we find it useful in the future, we can improve it later.